test/reducers/course.spec.js

```diff
@@ -1,16 +1,225 @@
 import deepFreeze from 'deep-freeze';
 import course from '../../app/assets/javascripts/reducers/course';
-import { UPDATE_COURSE } from '../../app/assets/javascripts/constants';
+import {
+  UPDATE_COURSE,
+  RECEIVE_COURSE,
+  RECEIVE_COURSE_UPDATE,
+  PERSISTED_COURSE,
+  CREATED_COURSE,
+  RECEIVE_INITIAL_CAMPAIGN,
+  ADD_CAMPAIGN,
+  RECEIVE_COURSE_CLONE,
+  DISMISS_SURVEY_NOTIFICATION,
+  TOGGLE_EDITING_SYLLABUS,
+  START_SYLLABUS_UPLOAD,
+  SYLLABUS_UPLOAD_SUCCESS,
+  LINKED_TO_SALESFORCE
+} from '../../app/assets/javascripts/constants';
 import '../testHelper';
 
 describe('course reducer', () => {
-  it('updates an attribute via UPDATE_COURSE', () => {
+  it('should return initial state when no action nor state is provided', () => {
+    const expectedState = {
+      title: '',
+      description: '',
+      school: '',
+      term: '',
+      level: '',
+      subject: '',
+      expected_students: '0',
+      start: null,
+      end: null,
+      timeline_start: null,
+      timeline_end: null,
+      day_exceptions: '',
+      weekdays: '0000000',
+      editingSyllabus: false
+    };
+
+    const newState = course(undefined, { type: null });
+    expect(newState).to.deep.eq(expectedState);
+  });
+
+  it('should handle RECEIVE_COURSE', () => {
+    const initialState = {};
+    const mockedAction = {
+      type: RECEIVE_COURSE,
+      data: { course: { title: 'title' } }
+    };
+    deepFreeze(initialState);
+
+    const newState = course(initialState, mockedAction);
+    expect(newState).to.deep.eq(mockedAction.data.course);
+  });
+
+  it('should handle RECEIVE_COURSE_UPDATE with newStats', () => {
```

Comment:
The point with this one should be to test the two possible cases: one where stats don't change and the current state is the same as the newly received data (which means newStats should equal false); and one where the stats do change (so newStats should equal true).

You probably want to test this with something like `{ title: 'title', createdCount: 0, editedCount: 0 }` as this initial state, and then for the action, update one of those and expect its newStats entry to be true, and don't update the other and expect it's newStats entry to be false.

Reply:
The description here could be more precise: it 'keeps track of which stats have updated with RECEIVE_COURSE_UPDATE' or something like that.